### PR TITLE
Logging Cleanup - Part 3

### PR DESCRIPTION
**Pull request by xWink**
**What type of PR is this?**

cleanup

**Which issue does this PR fix**:
#126 partially

**What does this PR do / Why do we need it**:
- Removes unused generated code
- Replace glog with Zap logger and improve logging clarity in service network and target group manager and service synthesizer
- Removes some instances of log n throw and unnecessary logs
- Adds check for errors before logging errors

**If an issue # is not available please add repro steps and logs from aws-gateway-controller showing the issue**:

**Testing done on this change**:

```
Ran 32 of 32 Specs in 2185.271 seconds
SUCCESS! -- 32 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestIntegration (2186.22s)
PASS
ok      github.com/aws/aws-application-networking-k8s/test/suites/integration   2186.939s
```

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

No

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this PR introduce any user-facing change?**:

Only logging changes

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by xWink** [comment]
Not sure why make presubmit keeps failing here when it passes locally. Am I missing a step?
```
aws-application-networking-k8s % make presubmit
?       github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1   [no test files]
ok      github.com/aws/aws-application-networking-k8s/pkg/aws   0.339s  coverage: 24.2% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/aws/services  0.358s  coverage: 5.8% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/config        0.538s  coverage: 32.9% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/deploy        0.708s  coverage: 20.6% of statements
?       github.com/aws/aws-application-networking-k8s/pkg/k8s   [no test files]
?       github.com/aws/aws-application-networking-k8s/pkg/model/lattice [no test files]
ok      github.com/aws/aws-application-networking-k8s/pkg/deploy/externaldns    1.083s  coverage: 75.6% of statements
?       github.com/aws/aws-application-networking-k8s/pkg/runtime       [no test files]
?       github.com/aws/aws-application-networking-k8s/pkg/utils [no test files]
?       github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog   [no test files]
?       github.com/aws/aws-application-networking-k8s/pkg/utils/retry   [no test files]
?       github.com/aws/aws-application-networking-k8s/pkg/utils/ttime   [no test files]
ok      github.com/aws/aws-application-networking-k8s/pkg/deploy/lattice        0.843s  coverage: 85.0% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/gateway       1.471s  coverage: 77.3% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/latticestore  0.847s  coverage: 64.1% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/model/core    1.689s  coverage: 61.7% of statements
ok      github.com/aws/aws-application-networking-k8s/pkg/model/core/graph      1.693s  coverage: 17.2% of statements
ok      github.com/aws/aws-application-networking-k8s/controllers       0.700s  coverage: 0.0% of statements
ok      github.com/aws/aws-application-networking-k8s/controllers/eventhandlers 1.027s  coverage: 53.5% of statements
kapshawn@88665a240c05 aws-application-networking-k8s % git status
On branch logging-3
nothing to commit, working tree clean
```

File: pkg/deploy/lattice/service_network_synthesizer.go [hunk 1]
```diff
@@ -44,34 +49,33 @@ func (s *serviceNetworkSynthesizer) Synthesize(ctx context.Context) error {
 	if ret != "" {
 		return errors.New(ret)
```
Comment (mikhail-aws) [inline pkg/deploy/lattice/service_network_synthesizer.go:50]: do you mind to cleanup this error handling too?

**Review by mikhail-aws — APPROVED** [review]

File: pkg/aws/gomock_reflect_3295326204/prog.go [hunk 2]
```diff
@@ -1,64 +0,0 @@
-package main
```
Comment (zijun726911) [inline pkg/aws/gomock_reflect_3295326204/prog.go:1]: Could you add these more `gomock_reflect_` in the gitignore in passing? 

https://github.com/aws/aws-application-networking-k8s/blob/3f89fd8c39c66f54af349eb9823fc8baf1a5944c/.gitignore#L18

Comment (xWink) [reply pkg/aws/gomock_reflect_3295326204/prog.go:1]: This shouldn't exist at all, so it shouldn't go into .gitignore, it was an artifact from a previous build issue that got checked in

File: pkg/deploy/lattice/service_network_synthesizer.go (hunk 1 above)
Comment (xWink) [reply pkg/deploy/lattice/service_network_synthesizer.go:50]: I gave this a try, but after a couple of hours of dealing with e2e test failures, reverted it for the sake of timeboxing. It seems as though we have some reliance on this error handling logic, which I'd like to keep outside the scope of the logging changes

**Pull request merged**